src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -25,25 +28,32 @@ const ContentTagsDropDownSelector = ({
   const [tags, setTags] = useState([]);
   const [nextPage, setNextPage] = useState(null);
 
-  // `fetchUrl` is initially `subTagsUrl` to fetch the initial data,
-  // however if it is null that means it is the root, and the apiHooks
-  // would automatically handle it. Later this url is set to the next
-  // page of results (if any)
-  //
-  // TODO: In the future we may need to refactor this to keep track
-  // of the count for how many times the user clicked on "load more" then
-  // use useQueries to load all the pages based on that.
-  const [fetchUrl, setFetchUrl] = useState(subTagsUrl);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  useEffect(() => {
+    if (searchTerm !== prevSearchTerm.current) {
+      setCurrentPage(1);
+    }
+  }, [searchTerm]);
 
   const isOpen = (i) => dropdownStates[i];
+  const closeAllDropdowns = () => {
+    const updatedStates = { ...dropdownStates };
+    // eslint-disable-next-line no-return-assign
+    Object.keys(updatedStates).map((key) => updatedStates[key] = false);
+    setDropdownStates(updatedStates);
+  };
 
   const clickAndEnterHandler = (i) => {
     // This flips the state of the dropdown at index false (closed) -> true (open)
     // and vice versa. Initially they are undefined which is falsy.
     setDropdownStates({ ...dropdownStates, [i]: !dropdownStates[i] });
   };
 
-  const { data: taxonomyTagsData, isSuccess: isTaxonomyTagsLoaded } = useTaxonomyTagsData(taxonomyId, fetchUrl);
+  const {
+    data: taxonomyTagsData,
+    isSuccess: isTaxonomyTagsLoaded,
+  } = useTaxonomyTagsData(taxonomyId, subTagsUrl, currentPage, searchTerm);
 
```

Comment:
@yusuf-musleh 
First of all, I think you generally are using `useEffect` too much - take a quick read of [You Might Not Need an Effect](https://react.dev/learn/you-might-not-need-an-effect#):

> If there is no external system involved (for example, if you want to update a component’s state when some props or state change), you shouldn’t need an Effect

So `useEffect` is not the right approach here.

This exact situation is explained in [Adjusting some state when a prop changes](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) so give that a read too. Although their primary recommendation would work but still may result in an extra fetch. So here's a way to avoid that:

### Complex solution

```
const [currentPageState, setCurrentPage] = useState(1);
const [currentPageFor, setCurrentPageFor] = useState(searchTerm);
let currentPage = currentPageState;
if (searchTerm !== currentPageFor) {
    // When the search term changes, reset to page 1
    setCurrentPage(1);
    setCurrentPageFor(searchTerm);
    // Apply this immediately to avoid a wasted fetch of later pages with the new search term
    currrentPage = 1;
}
```

### Simple solution

See [Resetting all state when a prop changes](https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes).